packages/main/src/plugin/provider-registry.ts

```diff
@@ -111,18 +111,32 @@ export class ProviderRegistry {
     this.lifecycleListeners = [];
     this.containerConnectionLifecycleListeners = [];
 
+    // Every 2 seconds, we will check:
+    // * The status of the providers
+    // * Any new warnings or informations for each provider
     setInterval(async () => {
       Array.from(this.providers.keys()).forEach(providerKey => {
+        // Get the provider and its lifecycle
         const provider = this.providers.get(providerKey);
         const providerLifecycle = this.providerLifecycles.get(providerKey);
+
+        // If the provider and its lifecycle exist, we will check
         if (provider && providerLifecycle) {
+          // Get the status
           const status = providerLifecycle.status();
+
+          // If the status does not match the current one, we will send a listener event and update the status
           if (status !== this.providerStatuses.get(providerKey)) {
             provider.updateStatus(status);
             this.listeners.forEach(listener => listener('provider:update-status', this.getProviderInfo(provider)));
             this.providerStatuses.set(providerKey, status);
           }
         }
+
+        // Let the renderer know to check for new warnings and information
+        if (provider) {
+          this.apiSender.send('provider:update-warning', provider.id);
+        }
```

Comment:
here, should we notify only if there is a change ? (like for this.providerStatuses.get(providerKey))

Reply:
I think that using update-warning may suffice as it doesn't use much computing cycles?

otherwise, I can go ahead and implement it via `registerLifecycle` + adding warnings()? what do you think.

Reply:
it will ask the client to refresh every 2 seconds the providers (as we send the order that warnings have been updated and that we refresh the providers when warnings are updated)

so I would avoid that using the form that you prefer.
Keeping the current warning per providers is trivial (like for the statuses)

We notify only if the JSON.stringify form is being changed

so it will refresh providers only when required and not every 2 seconds

Reply:
I would say it's the last blocker to merge the PR

Reply:
ah okay got it, i'll implement it like that